feat: add full screen ad hooks (#100)
invertase/react-native-google-mobile-ads: docs/displaying-ads-hook.mdx

# Hooks

> The AdMob package provides hooks to help you to display ads in functional component with tiny code. The supported ad formats are full-screen-type-ads: AppOpen, Interstitial, Rewarded.

Reviewer: ```suggestion
The AdMob package provides hooks to help you to display ads in a functional component with tiny code. The supported ad formats are full-screen ads: App open, Interstitial & Rewarded.
```

> ## Creating ad

Reviewer: ```suggestion
## Load an ad
```

You can create a new ad by adding a corresponding ad type's hook to your component.

The first argument of the hook is the "Ad Unit ID".
For testing, we can use a Test ID, however for production the ID from the
Google AdMob dashboard under "Ad units" should be used:

```tsx {4-8}
import { useInterstitialAd, TestIds } from 'react-native-google-mobile-ads';

export default function App() {
  const interstitialAd = useInterstitialAd(TestIds.Interstitial, {
    requestNonPersonalizedAdsOnly: true,
  });

  return <View>{/* ... */}</View>;
}
```

> > `adUnitid` parameter is also can be used to manage creation and destruction of ad instance.

Reviewer: ```suggestion
> The `adUnitid` parameter can also be used to manage creation and destruction of an ad instance.
```

> If `adUnitid` is set or changed, new ad instance will be created and previous ad instance will be destroyed if exists.
> If `adUnitid` is set to `null`, no ad instance will be created and previous ad instance will be destroyed if exists.

> The second argument is additional optional request options object to be sent whilst loading an advert, such as keywords & location.

Reviewer: ```suggestion
The second argument is an additional optional request options object to be sent whilst loading an advert, such as keywords & location.
```

Setting additional request options helps AdMob choose better tailored ads from the network. View the [`RequestOptions`](/reference/admob/requestoptions)
documentation to view the full range of options available.

> ## Displaying ad

Reviewer: ```suggestion
## Show the ad
```
